src/utils/DownloadUtils.ts

```diff
@@ -21,4 +28,66 @@ export function addSolutionsToError(err: R2Error): void {
 export function generateProgressPercentage(progress: number, currentIndex: number, total: number): number {
     const completedProgress = (currentIndex / total) * 100;
     return completedProgress + (progress * 1/total);
-}
\ No newline at end of file
+}
+
+
+export async function getFullDependencyList(
+    profile: ImmutableProfile,
+    combos: ThunderstoreCombo[],
+): Promise<ThunderstoreCombo[]> {
+    const modList = await ProfileModList.getModList(profile);
+    if (modList instanceof R2Error) {
+        throw modList;
+    }
+
+    const allModsToDownload: ThunderstoreCombo[] = [...combos];
+
+    for (const combo of combos) {
+        const isModpack = combo.getMod().getCategories().find(value => value === "Modpacks") !== undefined;
+        const versionMode = isModpack
+        ? DependencySetBuilderMode.USE_EXACT_VERSION
+        : DependencySetBuilderMode.USE_LATEST_VERSION;
+
+        await buildDependencySet(combo.getVersion(), allModsToDownload, versionMode);
+    }
+    sortDependencyOrder(allModsToDownload);
+
+    return allModsToDownload;
+}
+
+
+async function sortDependencyOrder(deps: ThunderstoreCombo[]) {
+    deps.sort((a, b) => {
+        if (a.getVersion().getDependencies().find(value => value.startsWith(b.getMod().getFullName() + "-"))) {
+            return 1;
+        } else {
+            return -1;
+        }
+    })
+}
+
+
+
+async function buildDependencySet(
+    mod: ThunderstoreVersion,
+    builder: ThunderstoreCombo[],
+    mode: DependencySetBuilderMode): Promise<void>
+{
+    const game = GameManager.activeGame;
```

Comment:
Would be a nice bonus if the game would be passed as an argument rather than depending on the GameManager, but that's not necessarily a matter for this PR.

Reply:
Any thoughts on this?

Reply:
I changed it to be passed as an argument, not too big of a change.
_Whats a few more to already a huge pile of changes xD._

Reply:
![pr-review](https://github.com/user-attachments/assets/a05aabbb-35b4-4484-be3c-eeb16d4dfbb5)